feat(compliance): add C5 Germany for aws

pedrooot opened the pull request:
### Description

This PR adds Cloud Computing Compliance Criteria Catalogue C5 compliance framework for the AWS provider
<img width="773" height="144" alt="Screenshot 2025-10-02 at 19 23 56" src="https://github.com/user-attachments/assets/380d9f37-88de-4bc4-b572-c1792e9f2ff0" />
<img width="3440" height="1290" alt="Screenshot 2025-10-02 at 19 24 15" src="https://github.com/user-attachments/assets/d2c83e59-a153-4383-a768-56fe21a83183" />

<img width="3124" height="909" alt="Screenshot 2025-10-02 at 19 25 05" src="https://github.com/user-attachments/assets/ec76f56d-ed34-4af3-b759-0715160fbdc6" />

<img width="1911" height="863" alt="Screenshot 2025-10-02 at 19 22 46" src="https://github.com/user-attachments/assets/0e351433-dd58-4769-8ea8-bb7d6da36978" />
<img width="806" height="616" alt="Screenshot 2025-10-02 at 19 30 40" src="https://github.com/user-attachments/assets/89b37928-cf9c-43af-8e18-d99bd5a1c47b" />

### Steps to review

Please add a detailed description of how to review this PR.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [ ] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.
- [ ] Review if is needed to change the [Readme.md](https://github.com/prowler-cloud/prowler/blob/master/README.md)
- [ ] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/prowler/CHANGELOG.md), if applicable.

#### API
- [x] Verify if API specs need to be regenerated.
- [x] Check if version updates are required (e.g., specs, Poetry, etc.).
- [x] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/api/CHANGELOG.md), if applicable.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

pedrooot committed:
feat(compliance): add C5 Germany WIP

pedrooot committed:
feat(compliance): add c5 germany support for AWS

pedrooot committed:
feat(export): change output order to keep consistency

pedrooot committed:
feat(ui): latest c5 touches

pedrooot committed:
chore(changelog): update with latest changes

pedrooot committed:
feat(ui): resolve conflicts from compliance-mapper.tsx

pedrooot committed:
Merge branch 'master' into PRWLR-8223-implement-c-5-compliance-for-aws

pedrooot committed:
fix(ui): tests and compliance version

pedrooot committed:
Merge branch 'PRWLR-8223-implement-c-5-compliance-for-aws' of github.com:prowler-cloud/prowler into PRWLR-8223-implement-c-5-compliance-for-aws

alejandrobailo commented on ui/components/compliance/compliance-accordion/compliance-accordion-requeriment-title.tsx:
```diff
@@ -7,12 +7,18 @@ interface ComplianceAccordionRequirementTitleProps {
 }
 
 export const ComplianceAccordionRequirementTitle = ({
+  type,
```
What is type?

alejandrobailo commented on ui/components/icons/compliance/IconCompliance.tsx:
```diff
@@ -1,6 +1,7 @@
 import AWSLogo from "./aws.svg";
```
Please run npm run lint:fix

alejandrobailo reviewed (commented):
Please fix the build, run the tests and take a look at the `compliance-accordion-requeriment-title`, if you can explain what is the type and attach an evidence

AdriiiPRodri reviewed (dismissed):
API side LGMT! 🚀

pedrooot committed:
Merge branch 'master' into PRWLR-8223-implement-c-5-compliance-for-aws

pedrooot committed:
fix(ui): lint

Alan-TheGentleman committed:
refactor(compliance): standardize requirement status handling with REQUIREMENT_STATUS constant

- Replace string literals ("PASS", "FAIL", "MANUAL") with REQUIREMENT_STATUS constant across all compliance mappers
- Simplify getComplianceIcon logic using object lookup instead of multiple if statements
- Extract accordion item creation into reusable functions in c5.tsx
- Improve c5-details component by using declarative sections array
- Enhance code maintainability and type safety for compliance status comparisons

pedrooot committed:
feat(ui): solve lint issues
